core/lib/multivm/src/versions/vm_latest/types/internals/transaction_data.rs

```diff
@@ -62,6 +62,12 @@ impl From<Transaction> for TransactionData {
                     U256::zero()
                 };
 
+                let should_deploy_contract = if execute_tx.execute.contract_address.is_none() {
```

Comment:
We should do it only if EVM-eq is enabled

Reply:
We can add the check to ensure it only happens with EVM equivalence enabled, but it might not be necessary due to the restriction [added here](https://github.com/lambdaclass/zksync-era/blob/2cec83f26e0b9309387135ca43718af4fcd6f6b1/core/lib/types/src/transaction_request.rs#L820-L823). Given this, no transaction should have a `None` contract address unless the EVM is enabled.